lib/rubocop/cop/style/hash_syntax.rb

```diff
@@ -60,7 +85,13 @@ def check(pairs, delim, msg)
               add_offense(pair,
                           pair.loc.expression.begin.join(pair.loc.operator),
                           msg) do
-                opposite_style_detected
+                if style == :dont_mix
+                  unrecognized_style_detected
+                else
+                  alternative_style = style == :ruby19 ? 'hash_rockets' : 'ruby19'
```

Comment:
Not sure what the best way to do this is... Really I'm trying to preserve the behaviour of `opposite_style_detected` (which expects 2 styles but here there are 3)